pkg/git/errors.go

```diff
@@ -0,0 +1,40 @@
+package git
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+)
+
+type gitError struct {
+	msg string
+}
+
+// GitError returns an error whereby the user part of the url is removed (e.g. an access token)
+// Accepts the message to use and the url to remove the user part from
+func GitError(msg, url string) error {
+	cleanedURL, err := cleanURL(url)
+	if err != nil {
+		return err
+	}
+	return gitError{msg: fmt.Sprintf("%s. repository URL = %s", msg, cleanedURL)}
+}
+
+// IsGitError checks if the error is a Git one (so there'll be a .User field in the URL)
+func IsGitError(err error) bool {
+	_, ok := err.(gitError)
+	return ok
+}
+
+func cleanURL(inputURL string) (string, error) {
```

Comment:
Can we have a small unit test for cleanURL()?

Reply:
`What's the latest on this @a-roberts ? Is err unsafe to print at this point?`

I haven't been able to get this to throw an error, even with obviously dodgy URLs (special chars in it, no https, no .com, etc). Interestingly if I try 

```
/services promote --from :https://github.com/a-roberts --to https://github.com__sds/a-roberts/staging --service service-a
```
notice the leading `:`, I get a go panic mentioning generateBranch instead

I'll work on adding a unit test for cleanURL

Reply:
Ok, got it erroring (had to change the code)

```
error for url.Parse is error for url.Parse is parse "https://github.com/a-roberts/myrepo\t___": net/url: invalid control character in URL
```

so, my token isn't printed, great

I changed our actual code to do

```
	destination, err := s.checkoutDestinationRepo("https://github.com/a-roberts/myrepo	___", newBranchName)
```